pkg/controller/controller.go

```diff
@@ -170,9 +168,10 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 }
 
 // worker runs a worker thread that just dequeues items, processes them, and marks them done.
-// If reconciler returns an error, requeue the item up to maxRetries before giving up.
+// If the reconciler returns that it should poll, requeue the item after the default polling interval elapses.
+// If the reconciler returns an error, requeue the item up to maxRetries before giving up.
 // It enforces that the reconciler is never invoked concurrently with the same key.
-func worker(queue workqueue.RateLimitingInterface, resourceType string, maxRetries int, reconciler func(key string) error) func() {
+func worker(queue workqueue.RateLimitingInterface, resourceType string, maxRetries int, reconciler func(key string) (bool, error)) func() {
```

Comment:
I thought we discussed adding another worker method so that we wouldn't have to change the reconcile functions for broker and binding.